packages/promise/test/intex.test.ts

```diff
@@ -75,6 +75,18 @@ describe("raceTimeout", () => {
     await vi.advanceTimersByTimeAsync(100);
     expect(done).toBe(reason);
   });
+
+  // test("rejects before timeout", async () => {
+  //   let done: unknown;
+  //   const reason = new Error("raceTimeout rejection reason");
+  //   const rejectetPromise = Promise.reject();  //vitest complains about unhandled promises, even though it is handled later
```

Comment:
There seems to be an issue involving rejected promises with vitest.
If the rejected promise isn't immediately handled with `then` or `catch`, vitest will show errors about unhandled errors after all tests, even though the test with rejected promise handles it later and test passes fully and successfully

Reply:
this is unfortunate.
tried this locally with the same effect